src/components/Suspense/SuspenseFallbackToPreviousContents.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { Suspense, useEffect, useRef } from 'react'
+
+/**
+ * This is useful for keeping the "last rendered" components on-screen while any suspense
+ * is triggered below this component.
+ *
+ * It stores a reference to the current children, and then returns them as the fallback.
+ */
+export const SuspenseFallbackToPreviousContents = (props: { children: React.ReactNode }) => {
```

Comment:
Would you consider `SuspenseWithLastRenderAsFallback`?

Reply:
i sort of like Previous overlap with usePrevious now that we made that change but don't mind much either way

Reply:
split the diff and used your structure with previous instead of last